### PR TITLE
feat: add "skip to content" navigation

### PR DESCRIPTION
**Pull request by mtwente**
# Pull request

## Proposed changes

* add a skip-to-content link that appears on focus
* add main content label to container structure
* add main content label to data story

See #105 

## Types of changes

- [x] New feature (non-breaking change which adds functionality).
- [ ] Enhancement (non-breaking change which enhances functionality)
- [ ] Bug Fix (non-breaking change which fixes an issue).
- [ ] Breaking change (fix or feature that would cause existing functionality to change).

## Checklist

- [x] I have read the **[README](./README.md)** document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

## Summary by CodeRabbit

- **New Features**
    - Improved accessibility by adding a "Skip to Main Content" link at the top of each page.
    - Enhanced page structure for better navigation and readability.

### CODE REVIEW

File: src/routes/data-stories/[...path]/+page.svelte [hunk 1]
```diff
@@ -7,5 +7,7 @@
 
 <Head />
 
+<main id="main-content"></main>
+
```
[the remark this answers is not in the document]
Comment (maehr) [reply src/routes/data-stories/[...path]/+page.svelte:11]: @mtwente I'd second coderabbit on this one.

File: src/lib/components/Container.svelte [hunk 2]
```diff
@@ -1,3 +1,5 @@
-<section class="prose p-4 lg:prose-xl lg:p-10" role="main">
-	<slot />
+<section class="prose p-4 lg:prose-xl lg:p-10">
+	<main id="main-content">
```
Comment (maehr) [inline src/lib/components/Container.svelte:2]: IMHO main should be around sections and articles
- https://developer.mozilla.org/en-US/docs/Web/HTML/Element/main?retiredLocale=de
- https://www.w3schools.com/tags/tag_main.asp

File: src/routes/+layout.svelte [hunk 3]
```diff
@@ -44,6 +44,12 @@
 	scrollbarGutter="auto"
 >
 	<svelte:fragment slot="header">
+		<div class="flex justify-center">
```
Comment (maehr) [inline src/routes/+layout.svelte:47]: Can you explain this feature on Thursday. Not yet fully grasping it.

**Review by maehr — CHANGES_REQUESTED** [review]
Thanks for the PR and enhancing ally features. We can go through it on Thursday.

File: src/routes/+layout.svelte (hunk 3 above)
Comment (mtwente) [reply src/routes/+layout.svelte:47]: This is just meant to make the skip-link appear centered on the page (and not cover the logo).

File: src/routes/data-stories/[...path]/+page.svelte (hunk 1 above)
[the remark this answers is not in the document]
Comment (mtwente) [reply src/routes/data-stories/[...path]/+page.svelte]: See 8fb4253.

File: src/lib/components/Container.svelte (hunk 2 above)
Comment (mtwente) [reply src/lib/components/Container.svelte]: See 7a2f991.

File: src/routes/data-stories/[...path]/+page.svelte (hunk 1 above)
[the remark this answers is not in the document]
Comment (maehr) [reply src/routes/data-stories/[...path]/+page.svelte]: Thanks for your commit. The ordering is fixed, but there is still two main tags in the resulting html
```html
<main id="page-content" class="flex-auto bg-white dark:bg-surface-900 border-t border-surface-500/10 "> 
  <main id="main-content">
     <section class="prose p-4 lg:prose-xl lg:p-10">
        <h1>Wir schreiben Basler Geschichte.</h1>
        <p><a href="/blog/buchvernissage-band-1-4-der-stadt-geschichte-basel"><strong data-svelte-h="svelte-2pug3j">Buchvernissage Band 1–4 der Stadt.Geschichte.Basel</strong></a><br> <strong>Montag, 11. März 2024, 18.30 Uhr<br>
           --
             | Historisches Museum Basel, Barfüsserkirche, Barfüsserplatz 7, Basel</strong>
        </p>
        <p>Die Webseite “Stadt.Geschichte.Basel” bietet einen tiefen Einblick in die facettenreiche Geschichte von Basel durch <a href="/baende" data-svelte-h="svelte-ygy2la">neun Einzelbände und einen Überblicksband</a>. Was sie besonders macht, ist der öffentliche Zugang zu Forschungsdaten, präsentiert als Data Stories. Plattform und Vermittlungsangebote sprechen ein breites Publikum an, von Geschichtsbegeisterten über Studierende bis Forschende, und laden dazu ein, Basels Geschichte auf innovative, datengetriebene Weise zu erkunden.</p>
        <p>Ergänzend dazu bietet der Blog einen Einblick in die Epochen <a href="/blog/kategorie/Ur-%20und%20Fr%C3%BChgeschichte" data-svelte-h="svelte-48kba5">Ur- und Frühgeschichte</a>, <a href="/blog/kategorie/R%C3%B6mische%20Zeit%20und%20Sp%C3%A4tantike" data-svelte-h="svelte-14kes2s">Römische Zeit und Spätantike</a>, <a href="/blog/kategorie/Mittelalter" data-svelte-h="svelte-38a7eg">Mittelalter</a>, <a href="/blog/kategorie/Neuzeit" data-svelte-h="svelte-877hx2">Neuzeit</a> und <a href="/blog/kategorie/Zeitgeschichte" data-svelte-h="svelte-12gwoky">Zeitgeschichte</a>.</p>
        <h2 id="data-stories"><a aria-hidden="true" tabindex="-1" href="#data-stories"><span class="icon icon-link"></span></a><a href="/data-stories" data-svelte-h="svelte-yz52qq">Data Stories</a></h2>
        <p data-svelte-h="svelte-1qhjf5v">Data Storytelling ist eine innovative Methode, bei der erzählerische Elemente und Daten miteinander kombiniert werden, um komplexe Informationen und Forschungsergebnisse in einer zugänglichen und ansprechenden digitalen Form zu präsentieren. Im Kontext von Stadt.Geschichte.Basel nutzen Autor*innen und Autoren-Kollektive diese Technik, um Geschichten und Forschungsdaten aus dem Projekt aufzubereiten und so die faszinierende Historie von Basel auf neue und spannende Weise zu vermitteln.</p>
        <ul>
           <li><a href="/data-stories/das-juedische-basel-1850-1914" data-svelte-h="svelte-1iodht1">Das jüdische Basel 1850–1914</a></li>
           <li><a href="/data-stories/das-kontrollbuero-und-die-wundermaschine" data-svelte-h="svelte-1bvbvwr">Das Kontrollbüro und die Wundermaschine: Wie Basel-Stadt in den 1960er Jahren das Einwohnermeldewesen digitalisierte</a></li>
        </ul>
        <p data-svelte-h="svelte-y3ugj4">Das Beste daran ist, dass immer neue Stories hinzukommen, sodass die Erkundung der Basler Geschichte nie endet.</p>
        <h2 id="forschungsdaten"><a aria-hidden="true" tabindex="-1" href="#forschungsdaten"><span class="icon icon-link"></span></a>Forschungsdaten</h2>
        <p data-svelte-h="svelte-ctsl83">Das Team Forschungsdatenmanagement entwickelt derzeit eine öffentlich zugängliche Plattform zur Bereitstellung von Quellen und historischen Forschungsdaten.</p>
        <figure class="flex flex-col items-center justify-center w-full">
           <img sizes="(max-width: 399px) 100vw, 399px" srcset="/files/2024/03/forschungsdatenplattform-399x500.jpg 399w, /files/2024/03/forschungsdatenplattform-240x300.jpg 240w, /files/2024/03/forschungsdatenplattform-768x961.jpg 768w, /files/2024/03/forschungsdatenplattform-1227x1536.jpg 1227w, /files/2024/03/forschungsdatenplattform-1636x2048.jpg 1636w, /files/2024/03/forschungsdatenplattform-1568x1963.jpg 1568w, /files/2024/03/forschungsdatenplattform-scaled.jpg 2045w" alt="Screenshot der Forschungsdatenplattform." src="/files/2024/03/forschungsdatenplattform.jpg" height="500" width="399" loading="lazy" decoding="async" class="mx-auto mb-2">
           <figcaption class="text-center" data-svelte-h="svelte-12kjd0k">Screenshot der Forschungsdatenplattform.</figcaption>
        </figure>
        <p data-svelte-h="svelte-75kttu">Auf dieser Forschungsdatenplattform werden Sie demnächst eine Sammlung an Ressourcen zur Geschichte der Stadt Basel finden können. Die Plattform bietet direkten Zugang zu nachnutzbaren Quellen und Daten aus den Bänden. Die Aufbereitung der Objekte ermöglicht es, bandübergreifend Themen aus der Basler Geschichte aus verschiedenen Blickwinkeln auf einer Karte, entlang einer Zeitachse oder nach Inhalten verschlagwortet zu entdecken. Weiterhin werden wissenschaftliche Publikationen im Zusammenhang mit dem Forschungsprojekt sowie die Bibliographie mit sämtlicher in den Bänden zitierter Literatur gesammelt zugänglich sein.</p>
        <h2 id="vermittlung"><a aria-hidden="true" tabindex="-1" href="#vermittlung"><span class="icon icon-link"></span></a>Vermittlung</h2>
        <p>Für die Vermittlung der neuen Stadtgeschichte arbeitet Stadt.Geschichte.Basel mit verschiedenen Kooperationspartner*innen zusammen. Die Veranstaltungen finden Sie unter <a href="/agenda" data-svelte-h="svelte-zhr0n7">Agenda</a>, die Kooperationspartner:innen unter <a href="/partner" data-svelte-h="svelte-19d6lsj">Partner</a>.</p>
        <h2 id="aktuelle-kooperationen"><a aria-hidden="true" tabindex="-1" href="#aktuelle-kooperationen"><span class="icon icon-link"></span></a>Aktuelle Kooperationen</h2>
        <p data-svelte-h="svelte-11rdn53">In der folgenden Tabelle finden Sie aktuelle Veranstaltungen sowie Vermittlungsangebote verschiedener Institutionen in Zusammenarbeit mit der Stadt.Geschichte.Basel.</p>
        <ul>
           <li><strong data-svelte-h="svelte-1b1tm10">📅 04.03.2024 🕒 18:15 Uhr, Historische Antiquarische Gesellschaft zu Basel (HAG)</strong>: <a href="/5750" data-svelte-h="svelte-1wv1ex5">Vorträge in der Alten Aula des Naturhistorischen Museums: Kevin Heiniger: “Patriot, Patriarch und Pietist? Facetten des Basler Tagebuchschreibers Joachim Weitnauer (1785-1848)”, Moritz Mähr: “Werkstattbericht zum neuen Portal Stadt.Geschichte.Basel”</a></li>
           <li><strong data-svelte-h="svelte-nsnmbk">📅 11.03.2024 🕒 18:30 Uhr, Historisches Museum Basel (HMB), Didi Offensiv</strong>: <a href="/blog/buchvernissage-band-1-4-der-stadt-geschichte-basel" data-svelte-h="svelte-13012ei">Vernissage</a> im HMB (Barfüsserkirche) mit Afterparty im Didi Offensiv</li>
           <li><strong data-svelte-h="svelte-1765pnj">📅 12.–16.03.2024 🕒 10:00–17.00 Uhr, HMB</strong>: Basler Geschichtstage: Parcours “Basel brennt!” für Schulklassen und Familien</li>
           <li><strong data-svelte-h="svelte-23g4p1">📅 12.–16.03.2024 🕒 10:00–17.00 Uhr, HMB, Christoph Merian Verlag (CMV)</strong>: Basler Geschichtstage: SGB-Info-Stand</li>
           <li><strong data-svelte-h="svelte-1farfi3">📅 13.03.2024 🕒 14:00–16:00 Uhr, Museum Kleines Klingental, Staatsarchiv Basel-Stadt (StaBS)</strong>: Führung “Wirtschaften mit dem Jenseits: Geschäftsmodelle der Klingentaler Klosterfrauen” mit Daniel Hagmann und Claudia Moddelmog</li>
           <li><strong data-svelte-h="svelte-1wca12">📅 13.03.2024 🕒 18:00–19:30 Uhr, Universitätsbibliothek Basel (UB), CMV</strong>: Vortrag und Gespräch mit Claudius Sieber-Lehmann: “Rauch über dem Wasser, Feuer am Himmel: Der Bischofshof wird 1247 abgefackelt”</li>
           <li><strong data-svelte-h="svelte-2lcqdu">📅 16.03.2024 🕒 10:00–17:00 Uhr, Verein Basler Geschichte, Volkshochschule beider Basel (VHSBB)</strong>: Basler Geschichtstage: Quiz zur Basler Geschichte</li>
           <li><strong data-svelte-h="svelte-mvrqcy">📅 29.04.2024 🕒 18:30–19:45 Uhr, VHSBB</strong>: Vortragsreihe “Ein Streiflicht auf die neue Buchreihe Stadt.Geschichte.Basel, Bände 1-4” an der Universität Basel oder per Zoom: Johannes Wimmer: “Neue Lebensbilder zum keltischen Basel. Band 1: Auf dem langen Weg zur Stadt”</li>
           <li><strong data-svelte-h="svelte-jqpkpz">📅 04.05.2024 🕒 14:00 Uhr, Frauenstadtrundgang</strong>: Premiere Stadtrundgang «Basel postkolonial»</li>
           <li><strong data-svelte-h="svelte-1czqnhc">📅 06.05.2024 🕒 18:30–19:45 Uhr, VHSBB</strong>: Vortragsreihe “Ein Streiflicht auf die neue Buchreihe Stadt.Geschichte.Basel, Bände 1-4”: Andreas Kettner: “Mit Karten durch die Basler Stadtgeschichte – Vom der Urlandschaft zur modernen Stadt”</li>
           <li><strong data-svelte-h="svelte-y1vz5e">📅 13.05.2024 🕒 18:30–19:45 Uhr, VHSBB</strong>: Vortragsreihe “Ein Streiflicht auf die neue Buchreihe Stadt.Geschichte.Basel, Bände 1-4”: Claudius Sieber-Lehmann: “Wie Efeu an der Bischofsmauer. Band 2: Eine Bischofsstadt zwischen Oberrhein und Jura”</li>
           <li><strong data-svelte-h="svelte-nfl7np">📅 17.05.2024 🕒 20:00–23:00 Uhr, Didi Offensiv</strong>: Pub-Quiz «Stadtgeschichte Basel»</li>
           <li><strong data-svelte-h="svelte-149cc33">📅 27.05.2024 🕒 18:30–19:45 Uhr, VHSBB</strong>: Vortragsreihe “Ein Streiflicht auf die neue Buchreihe Stadt.Geschichte.Basel, Bände 1-4”: Claudia Moddelmog: “Vom Neubau zum Umbau. Band 3: Stadt in Verhandlung”</li>
           <li><strong>📅 02.06.2024, Literatur<em data-svelte-h="svelte-tewf56">Spur</em></strong>: Literarischer Spaziergang mit Iris von Roten und Noëmi Crain-Merz</li>
           <li><strong data-svelte-h="svelte-1vzq4al">📅 03.06.2024 🕒 18:30–19:45, VHSBB</strong>: Vortragsreihe “Ein Streiflicht auf die neue Buchreihe Stadt.Geschichte.Basel, Bände 1-4”: Susanna Burghartz: “Vertraut und doch fremd. Band 4: Aufbrüche, Krisen, Transformationen”</li>
           <li><strong data-svelte-h="svelte-16cjn25">📅 September 2024, Museum.BL</strong>: Führung für Blinde und Sehbehinderte: “Eintauchen in die Welt des Seidenbands und der Heimposamenterei”</li>
           <li><strong data-svelte-h="svelte-krs8e7">📅 01.09.2024, HMB</strong>: Führungsreihe «Objekte schreiben Geschichte» an den Gratis-Sonntagen, September-Dezember</li>
           <li><strong>📅 08.09.2024, Literatur<em data-svelte-h="svelte-tewf56">Spur</em></strong>: Literarischer Spaziergang mit Iris von Roten und Noëmi Crain-Merz</li>
           <li><strong data-svelte-h="svelte-29wwkq">📅 25.09.2024, Abteilung Kultur Basel-Stadt</strong>: Mittwoch-Matinée: Jüdisches Museum</li>
           <li><strong data-svelte-h="svelte-1k0h2j6">📅 06.10.2024, HMB</strong>: Führungsreihe «Objekte schreiben Geschichte» an den Gratis-Sonntagen, September-Dezember</li>
           <li><strong data-svelte-h="svelte-ca1qr5">📅 09.10.2024 🕒 10:00–12:00 Uhr, Pharmaziemuseum und Abteilung Kultur Basel-Stadt</strong>: Mittwoch-Matinée am Pharmaziemuseum</li>
           <li><strong data-svelte-h="svelte-11n5o4r">📅 14.10.2024 🕒 18:15 Uhr, HAG</strong>: Vortragsreihe «Stadtgeschichte(n). Und jetzt?»: Podiumsdiskussion in Barfüsserkirche mit Isabel Koellreuter, Eva Herzog, Robi Labhardt, Lina Gafner, Moderation: David Tréfás</li>
           <li><strong>📅 15.10.2024, Literatur<em data-svelte-h="svelte-tewf56">Spur</em></strong>: Gesprächsrunde zu Iris von Rothen im Schmiedehof: Martina Kuoni und Noemi Crain Merz</li>
           <li><strong data-svelte-h="svelte-2f8dsc">📅 23.10.2024 🕒 10:00–12:00 Uhr, Archäologische Bodenforschung und Abteilung Kultur Basel-Stadt</strong>: Mittwoch-Matinée mit der Archäologischen Bodenforschung: Dagmar Bargetzi: Führung zur neugestalteten Informationsstelle “Römische Umfassungsmauer” im Hof des Schulhauses zur Mücke</li>
           <li><strong data-svelte-h="svelte-106qtuu">📅 28.10.2024 🕒 18:15 Uhr, HAG</strong>: Vortragsreihe «Stadtgeschichte(n). Und jetzt?»: mit Patrick Kury</li>
           <li><strong data-svelte-h="svelte-154bbxu">📅 30.10.2024 🕒 10:00–12:00 Uhr, Jüdisches Museum der Schweiz, Abteilung Kultur Basel-Stadt</strong>: Mittwoch-Matinée mit dem Jüdischem Museum: Patrick Kury und Barbara Häne</li>
           <li><strong data-svelte-h="svelte-wqh7ce">📅 01.11.2024, Schürch &amp; Koellreuter</strong>: Kulturstadtplan “Basel – Stadt der Tiere”: Isabel Koellreuter und Franziska Schürch: Kulturstadtplan zu Tieren und dem Verhältnis zwischen Menschen und Tieren durch die Jahrtausende</li>
           <li><strong data-svelte-h="svelte-j9yx4o">📅 03.11.2024, HMB</strong>: Führungsreihe «Objekte schreiben Geschichte» an den Gratis-Sonntagen, September-Dezember</li>
           <li><strong data-svelte-h="svelte-r9p9r1">📅 11.11.2024 🕒 18:15 Uhr, HAG</strong>: Vortragsreihe «Stadtgeschichte(n). Und jetzt?»: mit Stefan Hess</li>
           <li><strong data-svelte-h="svelte-t66vyc">📅 25.11.2024 🕒 18:15 Uhr, HAG</strong>: Vortragsreihe «Stadtgeschichte(n). Und jetzt?»: mit Gudrun Piller</li>
           <li><strong data-svelte-h="svelte-19c177">📅 01.12.2024, HMB</strong>: Führungsreihe «Objekte schreiben Geschichte» an den Gratis-Sonntagen, September-Dezember</li>
           <li><strong data-svelte-h="svelte-1bqpw1f">📅 09.12.2024 🕒 18:15 Uhr, HAG</strong>: Vortragsreihe «Stadtgeschichte(n). Und jetzt?»: (Informationen folgen)</li>
           <li><strong data-svelte-h="svelte-1858ry7">📅 13.01.2025 🕒 18:15 Uhr, HAG</strong>: Vortragsreihe «Stadtgeschichte(n). Und jetzt?»: Esther Baur</li>
           <li><strong data-svelte-h="svelte-3vjjvu">📅 27.01.2025 🕒 18:15 Uhr, HAG</strong>: Vortragsreihe «Stadtgeschichte(n). Und jetzt?»: Dominik Sieber</li>
           <li><strong data-svelte-h="svelte-1hz6kyh">📅 10.02.2025 🕒 18:15 Uhr, HAG</strong>: Vortragsreihe «Stadtgeschichte(n). Und jetzt?»: Céline Angehrn</li>
           <li><strong data-svelte-h="svelte-jesjg6">📅 24.02.2025 🕒 17:30 Uhr, HAG</strong>: Vortragsreihe «Stadtgeschichte(n). Und jetzt?»: Preis SWA</li>
        </ul>
     </section>
  </main>
</main>
```

It comes from the AppShell provided by Skeleton <https://www.skeleton.dev/components/app-shell>.

@mtwente I advise to remove your main tag and change the link to #page-content which is provided by skeleton.

File: src/routes/+layout.svelte [hunk 4]
```diff
@@ -44,6 +44,12 @@
...
+			<a
+				class="bg-secondary variant-filled-secondary btn absolute m-4 -translate-y-16 transition focus:translate-y-0"
+				href="#main-content">zum Seiteninhalt springen</a
```
Comment (maehr) [inline src/routes/+layout.svelte:50]:
```html
href="#page-content">zum Seiteninhalt springen</a
```

File: src/lib/components/Container.svelte [hunk 5]
```diff
@@ -1,3 +1,5 @@
-<section class="prose p-4 lg:prose-xl lg:p-10" role="main">
-	<slot />
-</section>
+<main id="main-content">
```
Comment (maehr) [inline src/lib/components/Container.svelte:1]:
```html
<section class="prose p-4 lg:prose-xl lg:p-10">
    <slot />
</section>
```

remove the main tag

File: src/routes/data-stories/[...path]/+page.svelte [hunk 6]
```diff
@@ -7,5 +7,12 @@
 
 <Head />
 
-<iframe src={data.url} class="h-full w-full" frameborder="0" title={config.title} allow="fullscreen"
-></iframe>
+<main id="main-content" class="h-full w-full">
```
Comment (maehr) [inline src/routes/data-stories/[...path]/+page.svelte:10]: Revert the changes

**Review by maehr — CHANGES_REQUESTED** [review]

File: src/lib/components/Container.svelte (hunk 5 above)
Comment (maehr) [reply src/lib/components/Container.svelte:1]: @mtwente revert this change as well

File: src/routes/+layout.svelte (hunk 4 above)
Comment (maehr) [reply src/routes/+layout.svelte:50]: Change the a tag href property accordingly.

**Review by maehr — CHANGES_REQUESTED** [review]
Thanks for the rework. There are still some things missing. IMHO the whole PR should only consist of the addition of a <a> tag to `+layout.svelte` and nothing more. `Container.svelte` does not need any changes.

**Comment by mtwente** [comment]
> Thanks for the rework. There are still some things missing. IMHO the whole PR should only consist of the addition of a tag to `+layout.svelte` and nothing more. `Container.svelte` does not need any changes.

I was interrupted last night before addressing the other errors 🙃 But now it should work, I think.

**Review by maehr — APPROVED** [review]

**Pull request merged**